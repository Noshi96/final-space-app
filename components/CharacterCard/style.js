import styled from 'styled-components'

export const Card = styled.div`
  border-radius: 24px;
  display: flex;
  align-items: center;
  flex-direction: column;
  gap: 1rem;
  padding-bottom: 1rem;
  text-align: center;
  overflow: hidden;
  cursor: pointer;
  transition: all 0.4s;
  width: 12rem;
  background: #ffffffba;

  &:hover {
    transform: scale(1.02);
  }

  @media only screen and (min-width: 48rem) {
    width: 16rem;
  }
`
export const Name = styled.div`
  display: inline-block;
  color: #333;
  font-size: 1rem;

  @media only screen and (min-width: 48rem) {
    font-size: 1.5rem;
  }
`

export const Image = styled.img`
  width: 100%;
`
