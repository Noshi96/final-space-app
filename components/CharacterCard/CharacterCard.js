import { Card, Name, Image } from './style'

const CharacterCard = ({ name, img_url }) => {
  return (
    <Card>
      <Image src={img_url} alt={name} />
      <Name>{name}</Name>
    </Card>
  )
}

export default CharacterCard
