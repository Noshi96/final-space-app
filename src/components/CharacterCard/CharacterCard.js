import { Card, Name, Image } from './style'
import useModal from 'hooks/useModal'
import CardDetailsModal from 'components/CardDetailsModal/CardDetailsModal'

const CharacterCard = ({ name, imgUrl, id, singleCharacter }) => {
  const { isOpen, toggle } = useModal()

  return (
    <>
      <Card onClick={toggle} tabIndex={0}>
        <Image src={imgUrl} alt={name} />
        <Name>{name}</Name>
      </Card>
      <CardDetailsModal
        isOpen={isOpen}
        closeModal={toggle}
        id={id}
        singleCharacter={singleCharacter}
      />
    </>
  )
}

export default CharacterCard
