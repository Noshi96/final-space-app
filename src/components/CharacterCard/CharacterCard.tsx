import { Card, Name, Image } from './style';
import useModal from 'hooks/useModal';
import CardDetailsModal from 'components/CardDetailsModal/CardDetailsModal';

interface CharacterCardProps {
  name: string;
  imgUrl: string;
  id: number;
  singleCharacter: any;
}

const CharacterCard = ({
  name,
  imgUrl,
  id,
  singleCharacter,
}: CharacterCardProps) => {
  const { isOpen, toggle } = useModal();

  return (
    <>
      <Card onClick={toggle} tabIndex={0}>
        <Image src={imgUrl} alt={name} />
        <Name>{name}</Name>
      </Card>
      <CardDetailsModal
        isOpen={isOpen}
        closeModal={toggle}
        id={id}
        singleCharacter={singleCharacter}
      />
    </>
  );
};

export default CharacterCard;
