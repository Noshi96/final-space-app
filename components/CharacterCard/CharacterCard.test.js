/* globals describe, it, expect  */

import { render } from '@testing-library/react'
import CharacterCard from './CharacterCard'

describe('CharacterCard', () => {
  const singleCharacter = {
    id: '1',
    name: 'Gary Goodspeed',
    imgUrl: 'https://finalspaceapi.com/api/character/avatar/gary_goodspeed.png',
  }

  it('renders character list', () => {
    const characterCard = render(
      <CharacterCard
        key={singleCharacter.id}
        name={singleCharacter.name}
        imgUrl={singleCharacter.img_url}
      />
    )
    expect(characterCard.container).toMatchSnapshot()
  })
})
