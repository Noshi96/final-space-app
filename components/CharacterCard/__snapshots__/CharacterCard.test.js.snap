// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard renders character card 1`] = `
<div>
  <div
    class="sc-7d7230a1-0 kNOnuu"
    tabindex="0"
  >
    <img
      alt="Gary Goodspeed"
      class="sc-7d7230a1-2 TDhcz"
      src="https://finalspaceapi.com/api/character/avatar/gary_goodspeed.png"
    />
    <div
      class="sc-7d7230a1-1 eLYphb"
    >
      Gary Goodspeed
    </div>
  </div>
</div>
`;
